src/main/java/org/example/bookapp/repository/book/AuthorSpecificationProvider.java

```diff
@@ -0,0 +1,24 @@
+package org.example.bookapp.repository.book;
+
+import java.util.Arrays;
+import org.example.bookapp.model.Book;
+import org.example.bookapp.repository.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthorSpecificationProvider implements SpecificationProvider<Book> {
+    private static final String FIRSTPARAM = "author";
```

Comment:
```suggestion
    private static final String AUTHOR_PARAM = "author";
```
class is named by author anyway, I think we can use it for constant name
not using _ makes it hard to read